gateway/reverse_proxy.go

```diff
@@ -385,15 +385,8 @@ type ReverseProxy struct {
 	Gw     *Gateway `json:"-"`
 }
 
-func (p *ReverseProxy) defaultTransport(dialerTimeout float64) *http.Transport {
-	timeout := 30.0
-	if dialerTimeout > 0 {
-		log.Debug("Setting timeout for outbound request to: ", dialerTimeout)
-		timeout = dialerTimeout
-	}
-
+func (p *ReverseProxy) defaultTransport() *http.Transport {
 	dialer := &net.Dialer{
-		Timeout:   time.Duration(float64(timeout) * float64(time.Second)),
```

Comment:
Since you removed default timeouts, we need to get them into the RoundTrip level. E.g. it is not safe to not have default timeout. 

Also we need to test really well if having timeout on request.context level is equal to having tiemout on the net.Dialer level.